src/plonk/verifier.rs

```diff
@@ -256,29 +256,18 @@ impl<'a, C: CurveAffine> Proof<C> {
         let x_7: C::Scalar = get_challenge_scalar(Challenge(transcript.squeeze().get_lower_128()));
 
         // Compute the final commitment that has to be opened
-        let mut f_commitment: C::Projective = self.f_commitment.to_projective();
+        let mut commitment_msm = params.empty_msm();
+        commitment_msm.add_term(C::Scalar::one(), self.f_commitment);
         for (_, &point_index) in srs.cs.rotations.iter() {
-            f_commitment *= x_7;
-            f_commitment = f_commitment + &q_commitments[point_index.0].as_ref().unwrap();
+            commitment_msm.scale(x_7);
+            commitment_msm.add_msm(&q_commitments[point_index.0]);
             f_eval *= &x_7;
             f_eval += &self.q_evals[point_index.0];
```

Comment:
we could consider renaming `f_eval` -> `commitment_eval`, since we've renamed `f_commitment` -> `commitment_msm`